openupgrade_scripts/scripts/project/14.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 ForgeFlow S.L.  <https://www.forgeflow.com>
+# Copyright 2021 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def _set_default_rating_status_period(env):
+    openupgrade.logged_query(
+        env.cr,
+        "UPDATE project_project SET rating_status_period = 'monthly' WHERE "
+        "rating_status_period IS NULL",
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.copy_columns(
+        env.cr,
+        {
+            "project_project": [
+                ("rating_status", None, None),
+            ],
+        },
+    )
+    openupgrade.rename_fields(
+        env,
+        [
+            (
+                "project.project",
+                "project_project",
+                "portal_show_rating",
+                "rating_active",
+            ),
+        ],
+    )
+    openupgrade.rename_xmlids(
+        env.cr,
+        [
+            ("project.access_partner_task user", "project.access_partner_task_user"),
+        ],
+    )
+    _set_default_rating_status_period(env)
+    # Add temporary tables for avoiding the automatic launch of the compute or default
+    openupgrade.logged_query(
+        env.cr,
+        "CREATE TABLE project_allowed_internal_users_rel (temp INTEGER)",
```

Comment:
CI says you should add the columns

Reply:
It seems Odoo has changed now the way it reflects the FK and indexes, so we only need to create the table at pre.